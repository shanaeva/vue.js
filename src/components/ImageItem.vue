<template>
  <div class="img-wrapper">
    <img
      class="img"
      :src="src"
      alt=""
    >
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'ImageItem',
  props: {
    src: {
      type: String,
      require: true,
      default: '',
    },
  },
});
</script>

<style scoped>
  .img-wrapper {
  width: 300px;
  height: 400px;
  border-radius: 4px;
  overflow: hidden;
  cursor: pointer;
  }
  .img {
  object-fit: cover;
  width: 100%;
  height: 100%;
  }
</style>
