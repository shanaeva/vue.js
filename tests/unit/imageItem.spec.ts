import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ImageItem from '@/components/ImageItem.vue';
import KillBill from '../../public/KillBill.jpg';

const intersectionObserverMock = () => ({
  observe: () => null,
  intersected: true,
});

window.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock);

describe('ImageItem', () => {
  const store = createStore({
    actions: {
      intersectionObserverMock,
    },
  });

  it('render', () => {
    const wrapper = shallowMount(ImageItem, { props: { src: KillBill }, store });
    expect(wrapper.props().src).toBe(KillBill);
  });
  it('render correctly', () => {
    const wrapper = shallowMount(ImageItem, { props: { src: KillBill } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
