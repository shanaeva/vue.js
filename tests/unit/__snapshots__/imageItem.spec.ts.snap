// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageItem render correctly 1`] = `
<div
  class="img-item"
>
  <img
    alt=""
    class="img"
    src=""
  />
</div>
`;
